src/main/java/roomescape/reservation/service/ReservationService.java

```diff
@@ -47,67 +47,54 @@ public ReservationService(
-            ReservationSaveRequest reservationSaveRequest,
-            LoginMember loginMember
-    ) {
-        Reservation reservation = createValidatedReservationOfStatus(reservationSaveRequest, loginMember, ReservationStatus.SUCCESS);
-        validateDuplicatedReservationSuccess(reservation);
-        Reservation savedReservation = reservationRepository.save(reservation);
-        if (loginMember.role().isUser()) {
-            paymentService.pay(PaymentRequest.from(reservationSaveRequest));
-        }
-
-        return ReservationResponse.toResponse(savedReservation);
-    }
-
-    public ReservationResponse saveReservationWaiting(
-            ReservationSaveRequest reservationSaveRequest,
-            LoginMember loginMember
-    ) {
-        Reservation reservation = createValidatedReservationOfStatus(reservationSaveRequest, loginMember, ReservationStatus.WAIT);
-        validateDuplicatedReservationWaiting(reservation, loginMember);
-        Reservation savedReservation = reservationRepository.save(reservation);
-
-        return ReservationResponse.toResponse(savedReservation);
-    }
-
-    private Reservation createValidatedReservationOfStatus(
-            ReservationSaveRequest reservationSaveRequest,
+    public ReservationResponse save(
+            UserReservationSaveRequest userReservationSaveRequest,
             LoginMember loginMember,
-            ReservationStatus reservationStatus
+            ReservationStatus status
     ) {
-        ReservationTime reservationTime = reservationTimeRepository.findById(reservationSaveRequest.getTimeId())
+        ReservationTime reservationTime = reservationTimeRepository.findById(userReservationSaveRequest.timeId())
                 .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 예약 시간입니다."));
 
-        Theme theme = themeRepository.findById(reservationSaveRequest.getThemeId())
+        Theme theme = themeRepository.findById(userReservationSaveRequest.themeId())
                 .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 테마입니다."));
 
         Member member = memberRepository.findById(loginMember.id())
                 .orElseThrow(() -> new IllegalArgumentException("존재하지 않는 회원입니다."));
 
-        return reservationSaveRequest.toReservation(member, theme, reservationTime, reservationStatus);
+        Reservation reservation = userReservationSaveRequest.toEntity(member, theme, reservationTime, status);
+
+        Reservation savedReservation = null;
+        if (status.isSuccess()) {
+            validateDuplicatedReservationSuccess(reservation);
+            savedReservation = reservationRepository.save(reservation);
+            paymentService.payForReservation(PaymentRequest.from(userReservationSaveRequest), savedReservation);
+        }
+        if (status.isWait()) {
+            validateReservationWait(loginMember, reservation);
+            savedReservation = reservationRepository.save(reservation);
+        }
```

Comment:
```suggestion
        validateReservation(loginMember, reservation);
        Reservation savedReservation = reservationRepository.save(reservation);
        if (status.isSuccess()) {
            paymentService.payForReservation(PaymentRequest.from(userReservationSaveRequest), savedReservation);
        }
```

이흐름으로 가고 validateReservaion 메서드 안에서 분기를 나눠 validate하는게 가독성 좋아보이네요

Reply:
레벨 1에서 학습했던 객체 지향 생활 체조의 "한 메서드에 오직 한 단계의 들여쓰기(indent)만 한다."를
지키기 위해 예약 확정과 대기 검증 로직을 분기 처리하지 않도 위와 같이 검증하도록 했는데
너무 위 규칙에 매몰되었나 봅니다 😅
주어진 요구사항도 아니였기도 하고, 제시해주신 방법대로 검증로직을 분기 처리하는 것이 더 가독성이 좋은 것 같네요 👍🏻